types/defines/pipeline-transform.d.ts

```diff
@@ -11,3 +11,18 @@ export abstract class PipelineTransform {
    */
   public transformJson(data: object[]): Promise<object[]>;
 }
+
+export type PipelineResponse = {
+  success: boolean
+  error?: string
+}
+
+export interface Pipeline {
+  /**
+   * send recieves an array of javascript objects which are sent
+   * into the pipeline for processing via binding
+   * @param data The data to be send
+   * @returns A promise containing the outcome of the send
+   */
+  send(data: object[]): Promise<PipelineResponse>
```

Comment:
Why an `object` - not every `object` is serializable? Does this cause issues with writing objects out to storage / consuming from a non-JS system?

How does this fit with Queues? https://github.com/cloudflare/workerd/blob/c9eca148301f683bf7b4bef2c642ef6380ea531e/src/workerd/api/queue.h#L51-L87

Reply:
This will act in the same way as a queue send in that the object must be structured cloneable since this is an RPC call. The only difference is the limit is 1mb instead of 128kb. https://developers.cloudflare.com/queues/configuration/javascript-apis/#queue

I think the idea behind using an object is that we want to make sure all the data in Pipelines is a row style layout. Rows can also be represented in other languages no problem and conversions of rows/columns to other formats is a well worn path.